src/middlewared/middlewared/plugins/cache.py

```diff
@@ -26,18 +27,36 @@ def get(self, key):
         Raises:
             KeyError: not found in the cache
         """
+        if isinstance(self.__cache[key][-1], datetime):
```

Comment:
That will break trying to get key that does not have timeout.

Reply:
It will raise a KeyError like the rest of the plugin does

Reply:
No, it could be TypeError, or getting the last char of a str,